yarn-project/sequencer-client/src/block_builder/solo_block_builder.ts

```diff
@@ -134,9 +134,8 @@ export class SoloBlockBuilder implements BlockBuilder {
     // Consolidate logs data from all txs
     const encryptedLogsArr: TxL2Logs[] = [];
     for (const tx of txs) {
-      if (tx.encryptedLogs) {
-        encryptedLogsArr.push(tx.encryptedLogs);
-      }
+      const logs = tx.encryptedLogs || new TxL2Logs([]);
```

Comment:
I set it up that way so that you can match encrypted logs to private txs by index in.

I also added check to the block so that it's ensured that num encrypted logs = num new commitments/4.